src/include/sof/audio/module_adapter/module/intel_modules.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+ */
+
+#ifndef __SOF_AUDIO_INTEL_MODULES__
+#define __SOF_AUDIO_INTEL_MODULES__
+
+#include <processing_module_adapter.h>
+
+/* Intel module adapter is an extension to SOF module adapter component that allows to integrate
+ * modules developed under IADK (Intel Audio Development Kit) Framework. IADK modules uses uniform
+ * set of interfaces and are linked into separate library. These modules are loaded in runtime
+ * through library_manager and then after registration into SOF component infrastructure are
+ * interfaced through module adapter API.
+ * Since IADK modules uses ProcessingModuleInterface to control/data transfer and AdspSystemService
+ * to use base FW services from internal module code, there is a communication shim layer defined
+ * in intel_module_adapter directory.
+ *
+ * Since ProcessingModuleInterface consists of virtual functions, there are C++ -> C wrappers
+ * defined to access the interface calls from SOF code.
+ *
+ * The main assumption here was to load IADK Modules without any code modifications. Therefore C++
+ * function, structures and variables definition are here kept with original form from
+ * IADK Framework. This provides binary compatibility for already developed 3rd party modules.
+ *
+ * There are three entities in intel module adapter package:
+ *  - System Agent - A mediator to allow the custom module to interact with the base SOF FW.
+ *                   It calls IADK module entry point and provides all necessary information to
+ *                   connect both sides of ProcessingModuleInterface and System Service.
+ *  - System Service - exposes of SOF base FW services to the module.
+ *  - Processing Module Adapter - SOF base FW side of ProcessingModuleInterface API
+ */
+
+
+struct comp_dev *intel_modules_shim_new(const struct comp_driver *drv,
```

Comment:
I think we want to keep ```intel``` here since this will be for Intel modules only - and not a general module API. i.e. changes to the API would be Intel driven.

Reply:
ok, but as long as we can spell this out in the Kconfig menu. i.e. so it's obvious that iadk is the Intel module API.

Reply:
names changed